salt/states/grains.py

```diff
@@ -144,17 +154,6 @@ def list_present(name, value, delimiter=DEFAULT_TARGET_DELIM):
             if set(value).issubset(set(__salt__['grains.get'](name))):
                 ret['comment'] = 'Value {1} is already in grain {0}'.format(name, value)
                 return ret
-            elif name in __context__.get('pending_grains', {}):
```

Comment:
Could you please clarify why removing this logic is necessary for achieving what the goal of this PR is? As I recall, this fixes another issue and removing it would cause a regression.